models/dataset.go

```diff
@@ -182,9 +182,6 @@ func CreateDataset(reader io.Reader) (*Dataset, error) {
 		return nil, errors.New("Failed to parse json body")
 	}
 
-	// Overwrite state to created
```

Comment:
if you get sent no `state`, should this now default `State` to `CreatedState`?  so, make this conditional:
`if dataset.State == "" {`...

Reply:
Good point - I have made the suggested change 👍